docs/release-guide.md

```diff
@@ -0,0 +1,97 @@
+# Release process
+
+This project uses [cargo-release][cargo-release] in order to prepare new releases, to tag and sign relevant git commit and to publish the resulting artifacts to [crates.io][crates-io].
+In order to ease downstream packaging of Rust binaries, [cargo-vendor][cargo-vendor] is also used to provide an archive of vendored dependencies (only relevant for offline builds).
+
+The release process follows the usual PR-and-review flow, allowing an external reviewer to have a final check before publishing.
+
+This document gives an high-level overview as well as a step-by-step guide on how to perform a release.
+
+## Overview
+
+Most of the process is automated with the help of cargo-release and its metadata entries in Cargo manifest.
+This helper is in charge of bumping the `version` field in the manifest in a dedicated commit, attaching the correspoding signed tag to it, and then producing another commit to prepare the project for the next development cycle.
+
+The two resulting commits can be then submitted in a dedicated PR and reviewed.
+Once merged, the last steps of the process consist in pushing the git tag, publishing the crate, and publishing the vendor tarball.
+
+## Steps
+
+If this is the first time, make sure all the requirements from the section below are met.
```

Comment:
If we move the `Requirements` section above the `Steps`, we won't need this statement.